src/test/java/org/hpccsystems/jdbcdriver/tests/HPCCDriverTest.java

```diff
@@ -278,49 +278,57 @@ private void executeFreeHandSQL(String sql, boolean expectPass, int minResults,
                 boolean success = true;
                 String[] csvlines = line.split(";");
                 prepParamValue = "";
-                for (int i = 0; i < csvlines.length; i++)
+                HPCCResultSet qrs = null;
+                long roundTripTimeMilli = 0;
+                try
                 {
-                    String value = null;
-                    String currline = csvlines[i];
-                    String [] valuesAndTheirTypes = currline.split(",");
-                    if (valuesAndTheirTypes.length > 0)
+                    for (int i = 0; i < csvlines.length; i++)
                     {
-                        value = String.valueOf(valuesAndTheirTypes[0]);
-                        if (valuesAndTheirTypes.length > 1)
-                            p.setObject(i + 1, HPCCJDBCUtils.deserializeSQLTypesToJava(valuesAndTheirTypes[1], value));
-                        else
-                            p.setObject(i + 1, value);
+                        String value = null;
+                        String currline = csvlines[i];
+                        String [] valuesAndTheirTypes = currline.split("[ ]*,[ ]*");
+                        if (valuesAndTheirTypes.length > 0)
```

Comment:
I think this might be unnecessary.  split should always return a String [] containing at least the original string.  Split would have thrown a nullptrexception otherwise.  I think we could safely set value=String.valueOf(valuesAndTheirTypes[0]); without this line.  I do think the check on 293 is necessary though, because the String array isn't guaranteed to have more than one element.